File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/entities/Newsletter.java
package abbruzzese.webpage.webpage.server.webpage.entities;

import java.util.UUID;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity
public class Newsletter {
    
    @Id
    private String id=UUID.randomUUID().toString();

    @Column(name="Email",nullable=false,unique=true)
    private String email;

    public Newsletter (){}

    Newsletter(String Email){
        this.email=Email;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

}

File: PHP_API/ENTITIES/Newsletter.php
<?php 

namespace App\Entity;

use Ramsey\Uuid\Uuid;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\NewsletterRepository;

#[ORM\Entity(repositoryClass: NewsletterRepository::class)] // Esto usa atributos de PHP 8
#[ORM\Table(name: "newsletter")]
class Newsletter{

    #[ORM\Id]
    #[ORM\Column(type:"string")]
    #[ORM\GeneratedValue(strategy:"NONE")]
    private $id;

    #[ORM\Column(type:"string",unique: true)]
    private $email;

    public function __construct($email) {
    $this->id = Uuid::uuid4()->toString();
    $this->email = $email;
    }

    //Getters y Setters
    public function getId() {return $this->id;}

    public function getEmail() {return $this->email;}
    public function setEmail($email) {$this->email;}

}


?>